pkg/controller/clusteringress/resources/route_test.go

```diff
@@ -1,19 +1,106 @@
 package resources
 
 import (
+	"errors"
 	"github.com/stretchr/testify/assert"
 	"testing"
 
+	"github.com/knative/serving/pkg/apis/networking"
 	networkingv1alpha1 "github.com/knative/serving/pkg/apis/networking/v1alpha1"
+	"github.com/knative/serving/pkg/apis/serving"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func TestMakeRoute(t *testing.T) {
+func TestMakeRouteWithNoRules(t *testing.T) {
 	ci := &networkingv1alpha1.ClusterIngress{
+		Spec: networkingv1alpha1.IngressSpec{
+			Visibility: networkingv1alpha1.IngressVisibilityExternalIP,
+		},
+	}
+
+	routes, err := MakeRoutes(ci)
+	assert.Zero(t, len(routes))
+	assert.Nil(t, err)
+}
+
+func TestMakeRouteInvalidHost(t *testing.T) {
```

Comment:
Is this testing for an invalid host (which I'd expect to be an error?) or just testing that we don't expose a route for an internal host?

Reply:
yes. Its testing that we don't expose route for an internal host

Because in this piece of code
https://github.com/openshift-knative/knative-openshift-ingress/blob/master/pkg/controller/clusteringress/resources/route.go#L33
we are checking for svc. 

I will change name from **TestMakeRouteInvalidHost** to **TestMakeRouteInternalHost**